editor/connections_dialog.cpp

```diff
@@ -788,23 +788,31 @@ bool ConnectionsDock::_is_item_signal(TreeItem &p_item) {
 }
 
 bool ConnectionsDock::_is_connection_inherited(Connection &p_connection) {
-	Node *scene_root = EditorNode::get_singleton()->get_edited_scene();
-	Ref<PackedScene> scn = ResourceLoader::load(scene_root->get_scene_file_path());
-	ERR_FAIL_NULL_V(scn, false);
-
-	Ref<SceneState> state = scn->get_state();
-	ERR_FAIL_NULL_V(state, false);
-
 	Node *source = Object::cast_to<Node>(p_connection.signal.get_object());
 	Node *target = Object::cast_to<Node>(p_connection.callable.get_object());
 
-	const NodePath source_path = scene_root->get_path_to(source);
-	const NodePath target_path = scene_root->get_path_to(target);
 	const StringName signal_name = p_connection.signal.get_name();
```

Comment:
```suggestion
	const StringName &signal_name = p_connection.signal.get_name();
```
StringNames should use references to avoid copying. Same with NodePaths.